// @aws-sdk/client-polly/dist-es/commands/StartSpeechSynthesisTaskCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartSpeechSynthesisTaskInputFilterSensitiveLog, StartSpeechSynthesisTaskOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1StartSpeechSynthesisTaskCommand, serializeAws_restJson1StartSpeechSynthesisTaskCommand, } from "../protocols/Aws_restJson1";
export class StartSpeechSynthesisTaskCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, StartSpeechSynthesisTaskCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "PollyClient";
        const commandName = "StartSpeechSynthesisTaskCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: StartSpeechSynthesisTaskInputFilterSensitiveLog,
            outputFilterSensitiveLog: StartSpeechSynthesisTaskOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_restJson1StartSpeechSynthesisTaskCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_restJson1StartSpeechSynthesisTaskCommand(output, context);
    }
}
